pulseapi/entries/views.py

```diff
@@ -26,77 +29,62 @@
 from pulseapi.utility.userpermissions import is_staff_address
 
 @api_view(['PUT'])
+@permission_classes([IsAuthenticated])
 def toggle_bookmark(request, entryid):
     """
     Toggle whether or not this user "bookmarked" the url-indicated entry.
     This is currently defined outside of the entry class, as functionality
     that is technically independent of entries themselves. We might
     change this in the future.
     """
-    user = request.user
 
-    if user.is_authenticated():
-        entry = None
+    user = request.user
```

Comment:
This issue is not about changing the code for bookmarking. Please don't overload PRs to do "more than they should". If you want to improve the bookmarking code, file a new issue for that and then a separate PR to effect the desired update.